test/state/HypermediaState.test.js

```diff
@@ -435,6 +435,40 @@ describe('HypermediaState class', () => {
 			});
 
 		});
+
+	});
+
+	it('pushes each state only once when there is a circular dependency', async() => {
+		// if this test fails with a timeout, it is likely that there is an infinite loop occurring
+		//   when we push the state due to the circular states
+
+		const entityHref = `http://entity-${uniqueId()}`;
+		fetchMock.mock(entityHref, {});
+		const observable = {
+			child: { observable: observableTypes.subEntity, rel: 'child', route: [{ efg: 'an additional route' }] }
+		};
+
+		const childEntity = {
+			class:['aClass'],
+			entities: [{
+				class: [ 'sub-class' ],
+				rel: [ 'child' ],
+				href: entityHref
+			}]
+		};
+
+		const state = await stateFactory(entityHref, 'token');
+		const observer = {};
+
+		state.addObservables(observer, observable);
+		await state.setSirenEntity(SirenParse(JSON.stringify(childEntity)));
+
+		const pushSpy = sinon.spy(state, 'push');
+
+		console.warn('if a timeout occurs here, it is due to an infinite loop from pushing to a circular dependency. Test in HypermediaState.test.js.');
+		await state.push();
+
+		assert(pushSpy.calledTwice);
```

Comment:
Can you put a note here after the assert so that we know why it's called twice? e.g., 
```suggestion
		assert(pushSpy.calledTwice, "Spy is called twice, where second call does nothing");
```